src/main/java/org/prebid/server/bidder/applogy/ApplogyBidder.java

```diff
@@ -0,0 +1,171 @@
+        final ExtImpApplogy extImpApplogy;
+        try {
+            extImpApplogy = Json.mapper.convertValue(imp.getExt().get("bidder"), ExtImpApplogy.class);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+
+        if (StringUtils.isBlank(extImpApplogy.getToken())) {
+            throw new PreBidException("token is empty");
+        }
+        return extImpApplogy;
+    }
+
+    private Imp processImp(Imp imp) {
+        Banner banner = imp.getBanner();
+        if (banner != null) {
+            if (banner.getH() == null && banner.getW() == null && CollectionUtils.isNotEmpty(banner.getFormat())) {
+                final Format firstFormat = banner.getFormat().get(0);
+                final Banner modifiedBanner = banner.toBuilder()
+                        .h(firstFormat.getH())
+                        .w(firstFormat.getW())
+                        .build();
+                return imp.toBuilder().banner(modifiedBanner).build();
+            }
+        }
+        return imp;
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        if (httpCall.getResponse().getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+        try {
+            final BidResponse bidResponse = Json.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+
+        } catch (DecodeException e) {
+            return Result.emptyWithError(BidderError.badServerResponse("failed to decode json"));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        final Map<String, BidType> requestImpIdToBidType = bidRequest.getImp().stream()
+                .collect(Collectors.toMap(Imp::getId, ApplogyBidder::getBidType));
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid,
+                        requestImpIdToBidType.getOrDefault(bid.getImpid(), BidType.banner), DEFAULT_BID_CURRENCY))
+                .collect(Collectors.toList());
+    }
+
+    private static BidType getBidType(Imp imp) {
+        if (imp.getVideo() != null) {
+            return BidType.video;
+        } else if (imp.getXNative() != null) {
```

Comment:
No need for `else if` when you have a return in previous condition. You can just keep 3 `if` statements, since each of them exits the method when `true`.

Also, you should throw/add an error when no bid type matched, just like they do in GO, and skip that bid. Check how I and @DGarbar did it in this PR - https://github.com/rubicon-project/prebid-server-java/pull/553/files in KubientBidder line 90 and 101, and check my second comment to his PR.

Basically, you need to pass an errors list and add an error to it whenever bid type was not matched, while returning a null instead of `BidderBid` object, so you can then filter these nulls out.